src/aegis128l/aegis128l.c

```diff
@@ -151,6 +151,26 @@ aegis128l_state_decrypt_detached_final(aegis128l_state *st_, uint8_t *m, size_t
     return implementation->state_decrypt_detached_final(st_, m, mlen_max, written, mac, maclen);
 }
 
+void
+aegis128l_randombytes_deterministic(uint8_t *out, size_t len, const uint8_t *npub, const uint8_t *k)
+{
+    implementation->randombytes_deterministic(out, len, npub, k);
+}
+
+void
+aegis128l_encrypt_unauthenticated(uint8_t *c, const uint8_t *m, size_t mlen, const uint8_t *npub,
+                                  const uint8_t *k)
+{
+    implementation->encrypt_unauthenticated(c, m, mlen, npub, k);
+}
+
+void
+aegis128l_decrypt_unauthenticated(uint8_t *m, const uint8_t *c, size_t clen, const uint8_t *npub,
+                                  const uint8_t *k)
+{
+    implementation->decrypt_unauthenticated(m, c, clen, npub, k);
+}
```

Comment:
Just curious, but is there any difference between the encrypt and decrypt functions?

IIUC the bit stream they generate are equal and they both do the same XOR.

Reply:
There is, because the plaintext is also added to the state. This is how the authentication tag is computed at the same time as encryption.

In the case authentication is not required, we could not inject anything, but the cipher stream would be different. And that would require changing the specification to include that specific and incompatible mode. 

It is easier and less confusing to assume that unauthenticated AEGIS is just regular AEGIS without the authentication tag.

Reply:
Thank you for the response.

I understand the reluctance to change the internals of libaegis. Also, I do not think there is any need to change the specification.

I was merely wondering if there is a need to provide both `encrypt_unauthenticated` and `decrypt_unauthenticated` when the observable behavior of the two functions are going to be identical to applications.

Reply:
The observable behavior is only identical for the first block :)